ml-algorithms/src/main/java/org/opensearch/ml/engine/utils/ZipUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Enumeration;
+
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
+import org.apache.commons.compress.archivers.zip.ZipFile;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * A util class contains zip file related operations.
+ */
+@Log4j2
+public class ZipUtils {
+
+    /**
+     * Uncompressed a zip file.
+     * @param zipFile zip file to be uncompressed
+     * @param dest the destination path of this uncompress
+     */
+    public static void unzip(File zipFile, Path dest) {
+        try {
+            ZipFile file = new ZipFile(zipFile);
+            Enumeration<ZipArchiveEntry> en = file.getEntries();
+            ZipArchiveEntry ze;
+            while (en.hasMoreElements()) {
+                ze = en.nextElement();
+                String name = ze.getName();
+                Path f = dest.resolve(name).toAbsolutePath();
+                if (!f.normalize().startsWith(dest))
+                    throw new RuntimeException("Bad zip entry");
```

Comment:
Can you write a test case to cover this branch?

Reply:
Never mind, I guess the edge case should be like this one https://codeql.github.com/codeql-query-help/java/java-zipslip/#recommendation

If zip entry has name ` ..\sneaky-file` , then we may unzip to a parent folder.

Reply:
Yes but actually this edge case is really hard to create, linux does not allow me to do that...